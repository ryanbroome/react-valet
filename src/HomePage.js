import React, { useContext } from "react";

// import Profile from "./Profile";
// import VehicleList from "./VehicleList";
// import LoginForm from "./LoginForm";

import ThemeContext from "./ThemeContext";

const HomePage = () => {
  // consume token from context, will be there if login, register or patch user
  const { token, userDetail } = useContext(ThemeContext);

  function loggedOut() {
    return (
      <div>
        <h1>Welcome To Park Pilot</h1>
      </div>
    );
  }
  //!  wrote this to try and get line 28 to work, for some reason right after login it can't read username or it's like userDetail.username doesn't work

  function loggedIn() {
    return (
      <div>
        <h1>Welcome To Park Pilot</h1>
        <p>Welcome new user: {userDetail ? userDetail.username : "Loading Spinner"}</p>
        <ul>
          TODO:
          <li>cleanup unused code and comments</li>
          <li>Update the size and shape of the elements to get them in position for a phone</li>
          <li>adjust layout to minimize buttons HOMEscreen, NAVBAR LOGIN REGISTER THEN NAVBAR BECOMES DROP DOWN HOMESCREEN ONLY BUTTONS NEEDED ADD VEHICLE / ACTIVE GARAGE</li>
          <li>update the background and colors</li>
          <li>write tests for all components and all functions</li>
        </ul>
      </div>
    );
  }

  return token ? loggedIn() : loggedOut();
};

export default HomePage;
