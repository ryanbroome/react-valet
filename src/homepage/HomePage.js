import React, { useContext } from "react";
import UserContext from "../auth/UserContext";

import ProfileDetail from "../profile/ProfileDetail";

const HomePage = () => {
  const { token, userDetail } = useContext(UserContext);

  function loggedOut() {
    return (
      <div>
        <h1>Welcome To Park Pilot</h1>
      </div>
    );
  }

  function loggedIn() {
    return (
      <div>
        <h1>Welcome To Park Pilot</h1>

        <div>
          {userDetail ? (
            <ProfileDetail userDetail={userDetail} />
          ) : (
            <div
              className="spinner-border text-primary"
              role="status">
              <span className="visually-hidden"></span>
            </div>
          )}
        </div>
      </div>
    );
  }

  return token ? loggedIn() : loggedOut();
};

export default HomePage;
