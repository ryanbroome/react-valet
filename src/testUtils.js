import React from "react";
import UserContext from "./auth/UserContext";
import App from "../src/App";

const MockUserContext = ({ children }) => {
  const mockToken = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VybmFtZSI6InRlc3R1c2VyIiwibG9jYXRpb25JZCI6MSwiaXNBZG1pbiI6ZmFsc2UsImlhdCI6MTcxMzA3NTM4Mn0.dXlnawfzk6wlsgpouoSp58QLvlOOSJFC03iq_FXab-I";

  const mockUserDetail = {
    userId: 1,
    vehicleId: 1,
    locationId: 1,
    transactionId: 1,
    ticketNum: "1",
    mobile: "5553334444",
    color: "red",
    make: "honda",
    username: "testuser",
  };

  const mockLocation = {
    sitename: "testSitename",
  };

  const mockVehicle = {
    ticketNum: 1,
    vehicleStatus: `parked`,
    mobile: "1113334444",
    color: "testColor",
    make: "testMake",
    damages: "testDamages",
    notes: "testNotes",
  };

  const mockTransaction = {
    userId: 1,
    vehicleId: 1,
    locationId: 1,
  };

  return <UserContext.Provider value={{ token: mockToken, userDetail: mockUserDetail, location: mockLocation, vehicle: mockVehicle, transaction: mockTransaction, update: App.update, register: App.register }}>{children}</UserContext.Provider>;
};

export { MockUserContext };
