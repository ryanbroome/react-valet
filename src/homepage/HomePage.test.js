import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router";
import HomePage from "./HomePage";
import { MockUserContext } from "../testUtils";

it("matches snapshot", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <MockUserContext>
        <HomePage />
      </MockUserContext>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});

it("matches snapshot when logged out", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <MockUserContext currentUser={null}>
        <HomePage />
      </MockUserContext>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
