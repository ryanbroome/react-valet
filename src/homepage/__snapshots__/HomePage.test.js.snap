// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h1>
      Welcome To Park Pilot
    </h1>
    <div>
      <div
        class="text-center card card-body bg-light"
      >
        <div
          class="card-body"
        >
          <h5
            class="card-title"
          >
            Current User 
          </h5>
          <h6
            class="card-subtitle"
          >
            testuser
          </h6>
          <ul
            class="list-group"
          >
            <li
              class="list-group-item"
            >
              Lifetime Parked: 
            </li>
            <li
              class="list-group-item"
            >
              Name:  
            </li>
            <li
              class="list-group-item"
            >
              Phone: 
            </li>
            <li
              class="list-group-item"
            >
              Email: 
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot when logged out 1`] = `
<DocumentFragment>
  <div>
    <h1>
      Welcome To Park Pilot
    </h1>
    <div>
      <div
        class="text-center card card-body bg-light"
      >
        <div
          class="card-body"
        >
          <h5
            class="card-title"
          >
            Current User 
          </h5>
          <h6
            class="card-subtitle"
          >
            testuser
          </h6>
          <ul
            class="list-group"
          >
            <li
              class="list-group-item"
            >
              Lifetime Parked: 
            </li>
            <li
              class="list-group-item"
            >
              Name:  
            </li>
            <li
              class="list-group-item"
            >
              Phone: 
            </li>
            <li
              class="list-group-item"
            >
              Email: 
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
